drivers/spi/spi_andes_atcspi200.c

```diff
@@ -195,23 +607,58 @@ static int transceive(const struct device *dev,
 			  spi_callback_t cb,
 			  void *userdata)
 {
+	const struct spi_atcspi200_cfg * const cfg = dev->config;
 	struct spi_atcspi200_data * const data = dev->data;
 	struct spi_context *ctx = &data->ctx;
 	int error, dfs;
+	size_t chunk_len;
 
 	spi_context_lock(ctx, asynchronous, cb, userdata, config);
-
 	error = configure(dev, config);
+
 	if (error == 0) {
 		data->busy = true;
 
 		dfs = SPI_WORD_SIZE_GET(ctx->config->operation) >> 3;
 		spi_context_buffers_setup(ctx, tx_bufs, rx_bufs, dfs);
 		spi_context_cs_control(ctx, true);
 
-		transfer_next_chunk(dev);
+		sys_set_bits(SPI_CTRL(cfg->base), CTRL_TX_FIFO_RST_MSK);
+		sys_set_bits(SPI_CTRL(cfg->base), CTRL_RX_FIFO_RST_MSK);
+
+		if (!spi_context_rx_on(ctx)) {
+			chunk_len = spi_context_total_tx_len(ctx);
+		} else if (!spi_context_tx_on(ctx)) {
+			chunk_len = spi_context_total_rx_len(ctx);
+		} else {
+			size_t rx_len = spi_context_total_rx_len(ctx);
+			size_t tx_len = spi_context_total_tx_len(ctx);
+
+			chunk_len = MIN(rx_len, tx_len);
+		}
+
+		data->chunk_len = chunk_len;
 
+#ifdef CONFIG_ANDES_SPI_DMA_MODE
+		if ((data->dma_tx.dma_dev != NULL)
+		 && (data->dma_rx.dma_dev != NULL)) {
```

Comment:
the new column limit is 100, this should fit:
```suggestion
		if ((data->dma_tx.dma_dev != NULL) && (data->dma_rx.dma_dev != NULL)) {
```